test/memkind_memtier_data_movement_test.cpp

```diff
@@ -706,6 +706,8 @@ TEST_F(PEBSTest, Basic)
     actual_size = ranking_get_total_size(mtt_allocator.internals.dramRanking);
     ASSERT_EQ(actual_size, size);
 
+    // Wait a little to allow PEBS thread do its work
+    sleep(2);
```

Comment:
Please, don't use ```sleep``` for thread synchronization, even if tests, unless really necessary.

There is a function that was written specifically for this purpose:
```C
    mtt_allocator_await_flush(&mtt_allocator);
```
Also, making the sleep non-interruptible significatly increases the execution time of any code that depends on flushing. Instead of calling ```sleep``` in pebs.c, I think we should call 
```C
        pthread_cond_timedwait(&mtt_allocator->bgThread.bg_thread_cond,
                          &mtt_allocator->bgThread.bg_thread_cond_mutex, wait_time);
```
in ```mtt_run_bg_thread```. This would automagically be interrupted by any call to ```mtt_allocator_await_flush``` (or ```mtt_allocator_await_balance```).

Reply:
```pthread_cond_clockwait``` would be more appropriate, in our case